packages/mobile/src/send/reducers.ts

```diff
@@ -48,13 +56,25 @@ export const sendReducer = (
   }
 }
 
+const sendPaymentOrInvite = (state: State, amount: BigNumber) => {
+  const latestPayment = { timestamp: Date.now(), amount }
+  const recentPayments = [...state.recentPayments, latestPayment]
```

Comment:
@jmrossy basically here we could filter out entries that are older than 24 hours. But my worry is would that violate the "pureness" of redux? If we pass in a timestamp through the action would that make it "more pure"?

Reply:
It's not perfect but I think calling for Date.now() here and filtering old ones is the most reasonable approach. Any others would probably be overly complex for no tangible benefit

Reply:
@jmrossy im surprised at that sense it would mean the reducer wouldnt be pure which I thought was pretty foundational to redux?

Reply:
@aaronmgdr I disagree. The pureness of a reducer is intended to mean that it has no side-effects other than the state modifications for actions it handles. This is just a state modification for a particular action, seems well within realm of intended use of reducers

Reply:
ah cool,

Reply:
yea I kept looking for examples of how other people handle time series or time based data structures but couldn't find anything serious. So, filter here? What about Aaron's suggestion to pass the timestamp from the saga in the action? That seems pretty good as well.

Reply:
One Direct benefit I see of passing in a timestamp is that it makes testing the reducer much simpler.